api/server/httputils/form.go

```diff
@@ -29,6 +29,29 @@ func BoolValueOrDefault(r *http.Request, k string, d bool) bool {
 	return BoolValue(r, k)
 }
 
+// Uint32Value parses a form value into an uint32 type. It returns an error
+// if the field is not set, empty, incorrectly formatted, or out of range.
+func Uint32Value(r *http.Request, field string) (uint32, error) {
+	// strconv.ParseUint returns an "strconv.ErrSyntax" for negative values,
+	// not an "out of range". Strip the prefix before parsing, and use it
+	// later to detect valid, but negative values.
+	v, isNeg := strings.CutPrefix(r.Form.Get(field), "-")
+	if v == "" || v[0] == '+' {
+		// Fast-path for invalid values.
+		return 0, strconv.ErrSyntax
+	}
+
+	i, err := strconv.ParseUint(v, 10, 32)
+	if err != nil {
+		// Unwrap to remove the 'strconv.ParseUint: parsing "some-invalid-value":' prefix.
+		return 0, errors.Unwrap(err)
+	}
+	if isNeg {
+		return 0, strconv.ErrRange
```

Comment:
FWIW; I was on the fence whether to make this function return an `errdefs.Invalid` on its own, but as we will likely always wrap the error to decorate it with some message ("invalid XYZ"), it would be a bit double.

Doing decorating as part of this function would've worked, using the `field` name that's passed, but in the current uses, those fields would be `w` or `h`, which would be a bit _too_ obscure for an error message, and I don't think we want to change the API to use `?width=xxx,height=xxx` _just_ for that purpose 😂 

We can still consider doing so at some point though (to prevent consumers forgetting to add a `errdefs` type.